library/init/meta/interactive.lean

```diff
@@ -6,8 +6,11 @@ Authors: Leonardo de Moura
 prelude
 import init.meta.tactic init.meta.rewrite_tactic init.meta.simp_tactic
 import init.meta.smt.congruence_closure init.category.combinators
+import init.meta.lean_parser
 
 namespace interactive
+@[reducible] meta def parse {α : Type} (p : lean_parser α) : Type := α
```

Comment:
This definition of parse requires a conversion from vm_objs to expr.  When we run p at parse time, it returns a vm_obj (for example of type `list expr`).  But then we want to insert the result in the pre-expression that we were originally parsing, so we need to convert it to expr.

We should just use `lean_parser expr` here, and make all the parsers construct pre-expressions by hand.